monomer.go

```diff
@@ -124,11 +123,9 @@ func (b *Block) ToEth() (*ethtypes.Block, error) {
 
 func (b *Block) ToCometLikeBlock() *bfttypes.Block {
 	return &bfttypes.Block{
-		Header: bfttypes.Header{
-			ChainID: b.Header.ChainID.String(),
-			Time:    time.Unix(int64(b.Header.Time), 0),
-			Height:  b.Header.Height,
-			AppHash: bftbytes.HexBytes(b.Header.AppHash),
+		Header: *b.Header.ToComet(),
+		Data: bfttypes.Data{
+			Txs: b.Txs,
```

Comment:
reminder to myself to move this to the next commit

Reply:
ultimately decided against this; would take an unnecessarily long time to do this. The real solution is to break up such large PRs :)